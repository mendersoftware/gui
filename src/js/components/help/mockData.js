// Copyright 2020 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {
    hasDeviceConfig: true,
    hasMonitor: true
  },
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '2003-01-01'
  }
};
