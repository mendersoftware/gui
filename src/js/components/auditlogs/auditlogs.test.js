// Copyright 2019 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import { LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';

import { prettyDOM } from '@testing-library/dom';
import { screen, render as testingLibRender, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { defaultState, undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import AuditLogs from './auditlogs';

const mockStore = configureStore([thunk]);

describe('Auditlogs Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({ ...defaultState, app: { ...defaultState.app, features: { ...defaultState.app.features, hasAuditlogs: true } } });
  });

  it('renders correctly', async () => {
    const { baseElement } = render(
      <LocalizationProvider dateAdapter={AdapterMoment}>
        <Provider store={store}>
          <AuditLogs />
        </Provider>
      </LocalizationProvider>
    );
    const view = prettyDOM(baseElement.firstChild, 100000, { highlight: false })
      .replace(/id="mui-[0-9]*"/g, '')
      .replace(/aria-labelledby="(mui-[0-9]* *)*"/g, '')
      .replace(/\\/g, '');
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });

  it('works as expected', async () => {
    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
    render(
      <LocalizationProvider dateAdapter={AdapterMoment}>
        <Provider store={store}>
          <AuditLogs />
        </Provider>
      </LocalizationProvider>
    );
    await user.click(screen.getByText(/last 7 days/i));
    await user.click(screen.getByText(/clear filter/i));
    await user.click(screen.getByRole('button', { name: /Download results as csv/i }));
    await user.click(screen.getByText(/open_terminal/i));
  });

  it('allows navigating by url as expected', async () => {
    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
    const ui = (
      <LocalizationProvider dateAdapter={AdapterMoment}>
        <MemoryRouter initialEntries={['/auditlog?startDate=2020-01-01']}>
          <Provider store={store}>
            <AuditLogs />
          </Provider>
        </MemoryRouter>
      </LocalizationProvider>
    );
    const { rerender } = testingLibRender(ui);
    await waitFor(() => rerender(ui));
    await user.click(screen.getByText(/clear filter/i));
  });
});
