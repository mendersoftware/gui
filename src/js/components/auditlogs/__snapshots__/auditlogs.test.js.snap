// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auditlogs Component renders correctly 1`] = `
"<div>
  <div
    class="fadeIn margin-left flexbox column"
    style="margin-right: 5%;"
  >
    <h3>
      Audit log
    </h3>
    <div
      class="auditlogs-filters margin-bottom margin-top-small css-1691o3y-filters"
    >
      <div
        class="MuiAutocomplete-root css-1h51icj-MuiAutocomplete-root"
        style="max-width: 250px;"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-lzl4b9-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1diiyoc-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="audit-log-user-selection"
            id="audit-log-user-selection-label"
          >
            Filter by user
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiAutocomplete-inputRoot css-1wsxxsk-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused css-15e8ec1-MuiInputBase-input-MuiInput-input"
              id="audit-log-user-selection"
              placeholder="Select a user"
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1h51icj-MuiAutocomplete-root"
        style="margin-left: 7.5px;"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-lzl4b9-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1diiyoc-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="audit-log-type-selection"
            id="audit-log-type-selection-label"
          >
            Filter by change
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1wsxxsk-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-15e8ec1-MuiInputBase-input-MuiInput-input"
              id="audit-log-type-selection"
              placeholder="Type"
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-1xk1zqq-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lqc678-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1h51icj-MuiAutocomplete-root"
        style="margin-right: 15px; margin-top: 16px;"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-lzl4b9-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1wsxxsk-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input Mui-disabled MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-15e8ec1-MuiInputBase-input-MuiInput-input"
              disabled=""
              id="audit-log-type-details-selection"
              placeholder="-"
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-1xk1zqq-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                disabled=""
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lqc678-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div />
      <div
        class=""
      >
        <span>
          Filter by date
        </span>
        <ul
          class="unstyled link-list horizontal"
        >
          <li>
            <a
              class="active"
            >
              Today
            </a>
          </li>
          <li>
            <a
              class=""
            >
              Yesterday
            </a>
          </li>
          <li>
            <a
              class=""
            >
              Last 7 days
            </a>
          </li>
          <li>
            <a
              class=""
            >
              Last 30 days
            </a>
          </li>
        </ul>
      </div>
      <div
        style="grid-column-start: 2; grid-column-end: 4; margin-left: 7.5px;"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1x6hzm-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1diiyoc-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="mui-2"
            id="mui-2-label"
          >
            From
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-s15sbk-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-15e8ec1-MuiInputBase-input-MuiInput-input"
              
              inputmode="text"
              placeholder="MMMM⁩ ⁦DD"
              type="text"
              value="January⁩ ⁦13th"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="Choose date, selected date is Jan 13, 2019"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-188h89-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lqc678-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root margin-left-small css-1x6hzm-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1diiyoc-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="mui-4"
            id="mui-4-label"
          >
            To
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-s15sbk-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-15e8ec1-MuiInputBase-input-MuiInput-input"
              
              inputmode="text"
              placeholder="MMMM⁩ ⁦DD"
              type="text"
              value="January⁩ ⁦13th"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="Choose date, selected date is Jan 13, 2019"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-188h89-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lqc678-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <span
        class="link margin-bottom-small"
        style="align-self: flex-end;"
      >
        clear filter
      </span>
    </div>
    <div
      class="flexbox center-aligned"
      style="justify-content: flex-end;"
    >
      <div
        class="loaderContainer shrunk"
      >
        <div
          class="  loader"
        >
          <span
            class="dot dot_1"
          />
          <span
            class="dot dot_2"
          />
          <span
            class="dot dot_3"
          />
          <span
            class="dot dot_4"
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1fxol34-MuiButtonBase-root-MuiButton-root"
        style="margin-left: 15px;"
        tabindex="0"
        type="button"
      >
        Download results as csv
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="fadeIn deploy-table-contain auditlogs-list"
    >
      <div
        class="auditlogs-list-item auditlogs-list-item-header muted"
      >
        <div
          class="columnHeader"
          style="cursor: initial;"
        >
          User
        </div>
        <div
          class="columnHeader"
          style="cursor: initial;"
        >
          Action
        </div>
        <div
          class="columnHeader"
          style="cursor: initial;"
        >
          Type
        </div>
        <div
          class="columnHeader"
          style="cursor: initial;"
        >
          Changed
        </div>
        <div
          class="columnHeader"
          style="cursor: initial;"
        >
          More details
        </div>
        <div
          class="columnHeader"
        >
          Time
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium sortIcon selected false css-1lqc678-MuiSvgIcon-root"
            data-testid="SortIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h6v-2H3v2zM3 6v2h18V6H3zm0 7h12v-2H3v2z"
            />
          </svg>
        </div>
        <div />
      </div>
      <div
        class="auditlogs-list"
      >
        <div
          class="auditlogs-list-item clickable"
        >
          <div>
            string@example.com
          </div>
          <div
            class="uppercased"
          >
            create
          </div>
          <div
            class="capitalized"
          >
            user
          </div>
          <div>
            user@acme.com
          </div>
          <div
            class="capitalized"
            style="align-items: flex-start; white-space: pre-line;"
          >
            change1
          </div>
          <time
            datetime="2019-01-01T12:10:22+00:00"
          >
            2019-01-01 12:10
          </time>
          <div
            class="uppercased link-color bold"
          >
            view details 
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lqc678-MuiSvgIcon-root"
              data-testid="ArrowRightAltIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
              />
            </svg>
          </div>
        </div>
        <div
          class="auditlogs-list-item clickable"
        >
          <div>
            string
          </div>
          <div
            class="uppercased"
          >
            create
          </div>
          <div
            class="capitalized"
          >
            deployment
          </div>
          <div>
            production
          </div>
          <a
            href="/deployments/finished?open=true&id=string"
          >
            View deployment
          </a>
          <time
            datetime="2019-01-01T12:16:22+00:00"
          >
            2019-01-01 12:16
          </time>
          <div
            class="uppercased link-color bold"
          >
            view details 
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lqc678-MuiSvgIcon-root"
              data-testid="ArrowRightAltIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
              />
            </svg>
          </div>
        </div>
        <div
          class="auditlogs-list-item clickable"
        >
          <div>
            string@example.com
          </div>
          <div
            class="uppercased"
          >
            open_terminal
          </div>
          <div
            class="capitalized"
          >
            device
          </div>
          <div>
            a1
          </div>
          <a>
            View session log
          </a>
          <time
            datetime="2019-01-01T12:10:22+00:00"
          >
            2019-01-01 12:10
          </time>
          <div
            class="uppercased link-color bold"
          >
            view details 
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lqc678-MuiSvgIcon-root"
              data-testid="ArrowRightAltIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="loaderContainer shrunk"
      >
        <div
          class="  loader"
        >
          <span
            class="dot dot_1"
          />
          <span
            class="dot dot_2"
          />
          <span
            class="dot dot_3"
          />
          <span
            class="dot dot_4"
          />
        </div>
      </div>
      <div
        class="MuiTablePagination-root flexbox margin-top  css-1s8wd9s-MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            class="MuiTablePagination-spacer flexbox no-basis css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            class="MuiTablePagination-selectLabel css-1ecnf2s-MuiTablePagination-selectLabel"
            
          >
            Rows
          </p>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary css-7kd4nt-MuiInputBase-root-MuiTablePagination-select"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              
              class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
              
              role="button"
              tabindex="0"
            >
              20
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="20"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-dwvfc1-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTablePagination-displayedRows css-cmytl3-MuiTablePagination-displayedRows"
          />
          <div
            class="flexbox center-aligned"
          >
            <div>
              1-3 of 3
            </div>
            <button
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-13hun5h-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lqc678-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                />
              </svg>
            </button>
            <div>
              <button
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-13hun5h-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lqc678-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
